python/tests/test_account.py

```diff
@@ -1402,6 +1402,47 @@ def ac_member_removed(self, chat, contact, message):
         assert ev.action == "removed"
         assert ev.message.get_sender_contact().addr == ac1_addr
 
+    def test_system_group_msg_from_blocked_user(self, acfactory, lp):
+        """
+        Tests that a blocked user removes you from a group.
+        The message has to be fetched even though the user is blocked
+        to avoid inconsistent group state.
+        Also tests blocking in general.
+        """
+        lp.sec("Create a group chat with ac1 and ac2")
+        (ac1, ac2) = acfactory.get_two_online_accounts()
+        acfactory.introduce_each_other((ac1, ac2))
+        chat_on_ac1 = ac1.create_group_chat("title", contacts=[ac2])
+        chat_on_ac1.send_text("First group message")
+        chat_on_ac2 = ac2._evtracker.wait_next_incoming_message().chat
+
+        lp.sec("ac1 blocks ac2")
+        contact = ac1.create_contact(ac2)
+        contact.set_blocked(True)
+        assert contact.is_blocked()
+
+        lp.sec("ac2 sends a message to ac1 that does not arrive because it is blocked")
+        ac2.create_chat(ac1).send_text("This will not arrive!")
+        with pytest.raises(queue.Empty):
+            ac1._evtracker.get_matching("DC_EVENT_INCOMING_MSG", timeout=10)
```

Comment:
This can be removed: since ac1 will wait for "This will arrive" later, it will be an error if this message arrives first.

Currently it introduces an unnecessary 10-seconds delay which is bad for overall testing time (even if you we run multiple tests in parallel).